[IV-25-26] Objetivo 3

chelunike opened the pull request:
* [x] ¿Incluye el fichero `iv.yaml` las claves de todos los objetivos
anteriores?
* [x] ¿Se ha desarrollado el fichero para las tareas a partir de un
   issue?
* [x] ¿Ese issue está asignado a un milestone?
* [x] ¿El milestone incluye, entre sus productos mínimamente viables, la
  existencia de herramientas que permitan automatizar tareas?
* [x] ¿Hemos justificado correctamente en caso de elegir un gestor de
   dependencias el que se use como gestor de tareas?
* [x] ¿El PR, que también es un issue, está asignado al mismo milestone?
* [x] ¿La tarea `check` efectivamente comprueba la sintaxis de los fuentes
  existentes y sólo eso?
* [x] ¿Se ha comprobado que las instrucciones incluidas en el `README.md` son
completas y correctas?
Todo esto en relación a #14

oscar0310 committed:
fix #6 Añadiendo lenguaje

Se añade el lemguaje que se va a usar en el proyecto

oscar0310 committed:
fix #8 añadiendo directorio lib

oscar0310 committed:
Fix #9 Añadiendo el archivo CVE.pm

Representará los CVE

oscar0310 committed:
Fix #9 Cambiando nombre  CVE.pm a cve.pm

Siguiendo las buenas prácticas.

oscar0310 committed:
Fix #9 Añadiendo cve.pm

oscar0310 committed:
fix #9 añadienco la clase cve

oscar0310 committed:
fix #8 gitignore de acuerdo con Perl

Se ha actualizado el gitignore para Perl

oscar0310 committed:
fix #9 eliminando cve.pm closes #9

oscar0310 committed:
Cerrando issue no valido Closes #9

oscar0310 committed:
fix #11 añadiendo entidad ctf

oscar0310 committed:
fix #11 creando ctf.pm

oscar0310 committed:
fix #11 añadiendo tipo a ctf

contiene el tipo de reto CTF

oscar0310 committed:
fix #11 añadiendo ip  a ctf

contiene la dirección ip del sistema de el reto CTF

oscar0310 committed:
fix #11 añadiendo port a ctf

contiene el puerto  del sistema de el reto CTF

oscar0310 committed:
fix #12 añadiendo la entidad cve

chelunike committed:
Añadida descripción de la elección del gestor de tareas y de dependencias close #14
Nos hemos decidido por usar Make y Carton + cpanfile + CPAN, respectivamente.

chelunike committed:
Añadida descripción de automatización de tareas
Dado que hemos elegido make como se pide en  #14

chelunike committed:
Añadido enlace a la documentación de elección de gestores
Enlace que lleva a la información que se pide en #14

chelunike committed:
Eliminando opciones de ExtUtils::MakeMaker del gitignoer
Puesto que al final no lo usamos según lo resuelto para #14

chelunike committed:
Añadido fichero de configuración para make
Según lo resuelto para #14 he añadido el fichero Makefile según los requisitos del objetivo

chelunike committed:
Añadido fichero de configuración para Carton
Según la solución escogida para #14, he añadido el fichero cpanfile vacio para que se ejecute sin problemas las tareas del objetivo a través de make sin que de error.

chelunike committed:
Añadida instrucciones de uso del gestor de tareas para #14

oscar0310 commented on docs/gestores.md:
```diff
@@ -0,0 +1,75 @@
+
+## Gestor de Dependencias
+
+Tras revisar todos las posibles opciones de gestión de dependencias con Perl, en este lenguaje hay muchas formas de hacer las cosas. Y eso crea un monto de formas de gestionar las dependencias, en este caso tenemos que diferenciar entre empaquetadores, la propia herramienta que instala las dependencias y el formato con el que se empaquetan dichas dependencias.
```
```suggestion
Tras revisar todos las posibles opciones de gestión de dependencias con Perl, en este lenguaje hay muchas formas de hacer las cosas. Y eso crea un montón de formas de gestionar las dependencias, en este caso tenemos que diferenciar entre empaquetadores, la propia herramienta que instala las dependencias y el formato con el que se empaquetan dichas dependencias.
```

oscar0310 commented on docs/gestores.md:
```diff
@@ -0,0 +1,75 @@
+Tras revisar todos las posibles opciones de gestión de dependencias con Perl, en este lenguaje hay muchas formas de hacer las cosas. Y eso crea un monto de formas de gestionar las dependencias, en este caso tenemos que diferenciar entre empaquetadores, la propia herramienta que instala las dependencias y el formato con el que se empaquetan dichas dependencias.
+
+Nuestras prioridades de cara a la gestión de dependencias son las siguientes:
+- Que funcione para nuestro lenguaje Perl
```
¿Crees que es necesario este criterio?

oscar0310 commented on docs/gestores.md:
```diff
@@ -0,0 +1,75 @@
+
+Nuestras prioridades de cara a la gestión de dependencias son las siguientes:
+- Que funcione para nuestro lenguaje Perl
+- Facilidad y practica respecto al uso e instalación para su despliegue.
```
¿Cómo mides la facilidad ? y  ¿es objetivo ?.

oscar0310 commented on docs/gestores.md:
```diff
@@ -0,0 +1,75 @@
+Nuestras prioridades de cara a la gestión de dependencias son las siguientes:
+- Que funcione para nuestro lenguaje Perl
+- Facilidad y practica respecto al uso e instalación para su despliegue.
+- Nos asegure una estabilidad en los paquetes y soporte
```
¿En que se mide que sea estable?

oscar0310 commented on docs/gestores.md:
```diff
@@ -0,0 +1,75 @@
+## Gestor de tareas
+
+Respecto al gestor de tareas, tenemos unas prioridades parecidas.
+- Facilidad de uso e instalación
```
Igual que en el gestor de dependencias.

oscar0310 commented on docs/gestores.md:
```diff
@@ -0,0 +1,75 @@
+Respecto al gestor de tareas, tenemos unas prioridades parecidas.
+- Facilidad de uso e instalación
+- Nos permitan el desarrollo de tests para futuros objetivos
+- Herramienta estable y que no genere demasiada deuda técnica
```
En cuanto a la estabilidad igual que en el gestor de dependencias y sobre la deuda técnica, ¿ se puede medir cómo tal?, La deuda técnica influye a la decisión , pero son otros aspectos que sí son objetivos y medibles relacionados con la deuda técnica los que tienes que usar.

chelunike committed:
Corrigiendo typo en docs/gestores.md como se menciona en #15

Co-authored-by: Óscar Fernández Rodríguez <e.oscar12@go.ugr.es>

oscar0310 reviewed (commented):
Buenas @chelunike he estado revisando tu objetivo 3, he visto que has hecho un gran trabajo investigando sobre los gestores de dependencias y tareas de Perl, te he hecho una serie de comentarios sobre los criterios  que usas para escoger la herramienta.

chelunike committed:
Eliminando criterio redundante como es el lenguaje
Esto se comenta en la PR #15 solucionando el issue #14

chelunike committed:
Modificando prioridades de elección
Tal como se ha comentado en #15 para solucionar el issue #14

chelunike committed:
Merge branch 'Objetivo-3' of git:chelunike/didactic-chainsaw into Objetivo-3

GabrielFranciscoSM commented on docs/gestores.md:
```diff
@@ -0,0 +1,74 @@
+- Aislamiento de los paquetes para evitar futuros errores 
+
+### Opciones encontradas
+A continuación, revisamos varias opciones viables. Todas ellas son proyectos estables y con soporte actualmente.
```
Si todos son proyectos estables y con soporte actual, ¿Realmente lo estas usando como criterio de elección?

GabrielFranciscoSM commented on docs/gestores.md:
```diff
@@ -0,0 +1,74 @@
+#### CPAN
+- Facilidad de instalación y uso puesto que viene con Perl
+- Instala las dependencias de manera global, sin aislarlas respecto al proyecto 
+- No gestiona lista de dependencias ni versiones, es una herramienta para instalar las dependencias
```
¿Esto es parte de los criterios?

GabrielFranciscoSM commented on docs/gestores.md:
```diff
@@ -0,0 +1,74 @@
+
+#### CPANMINUS
+Como CPAN, sirve solo para instalar dependencias
+- Fácil de usar
```
¿Cómo mides que sea fácil de usar? ¿Es un criterio objetivo?

GabrielFranciscoSM commented on docs/gestores.md:
```diff
@@ -0,0 +1,74 @@
+
+### Conclusión
+
+Usamos -> Carton + cpanfile + Cpan
```
Si vas a usar más de una herramienta como "gestor de dependencias", ¿No deberías evaluarlas en su conjunto? ¿Tiene sentido evaluar unas trente a otras para después elegir varias?

GabrielFranciscoSM commented on docs/gestores.md:
```diff
@@ -0,0 +1,74 @@
+
+#### Task
+- Tendríamos que instalar JS, para usarla, lo cual nos dificulta inicialmente en la instalación.
+- Es viable para gestionar las tareas de este proyecto
```
Cualquier gestor de tareas cumple esto, ¿no?

GabrielFranciscoSM commented on docs/gestores.md:
```diff
@@ -0,0 +1,74 @@
+### Conclusión
+
+Usamos -> Make
+Por estabilidad y dada que es una herramienta comúnmente usada en nuestro entorno, nos será más fácil de usar.
```
> comúnmente usada en nuestro entorno

¿Es justo valorar una herramienta con criterios que no has definido antes?

GabrielFranciscoSM commented on docs/gestores.md:
```diff
@@ -0,0 +1,74 @@
+
+Respecto al gestor de tareas, tenemos unas prioridades parecidas.
+- Resulte fácil de instalar y usar para el programador; es decir, que en pocos pasos pueda lanzar las tareas.
+- Nos permitan el desarrollo de tests para futuros objetivos
```
¿Usas este criterio en las diferentes opciones?

GabrielFranciscoSM reviewed (commented):
No entiendo muy bien las herramientas de gestión de dependencias, ya que al final usas todas menos cpanminus.

chelunike committed:
Separadas las herramientas de gestión de dependencias
Tal y como hemos comentado en #15 para una entender mejor la elección de las herramientas. Para solucionar #14

chelunike commented:
Revisión @GabrielFranciscoSM  @oscar0310

chelunike commented:
Revisión @jorgelopez-ugr

jorgelopez-ugr commented on docs/gestores.md:
```diff
@@ -0,0 +1,73 @@
+Tras revisar todos las posibles opciones de gestión de dependencias con Perl, en este lenguaje hay muchas formas de hacer las cosas. Y eso crea un montón de formas de gestionar las dependencias, en este caso tenemos que diferenciar entre empaquetadores, la propia herramienta que instala las dependencias y el formato con el que se empaquetan dichas dependencias.
+
+Nuestras prioridades de cara a la gestión de dependencias son las siguientes:
+- Resulte fácil de instalar y usar para el programador; es decir, que en pocos pasos pueda disponer de las dependencias.
```
Creo que este criterio es dificil de medir, ¿cómo determinas cuál es fácil y cuál es dificil?¿Como determinas cuántos son "pocos pasos"? Intenta expresar los criterios como cosas 100% objetivas.

jorgelopez-ugr reviewed (commented):
Te he comentado lo de los criterios objetivos y creo que eso condicionará bastante el resto de tu justificación por lo que en principio no te añado más comentarios.

Revisa la propuesta para ajustarla. :+1:

vpedrosa commented on docs/gestores.md:
```diff
@@ -0,0 +1,73 @@
+Y en segundo lugar, vamos a ver que herramienta podemos usar para poder aislar la instalación de los paquetes a nuestro proyecto. Estás se sirven de las anteriores, por lo que la elección es entre las siguientes:
+
+#### Carton
+Se trata de un empaquetador (Bundler) y empaqueta todo a través de un fichero con formato `cpanfile` (No se refiere a una caja de cartón :)
```
Uno de los errores comunes es "Evitar información superflua y enfocarse en criterios que justifiquen objetivamente la elección". Creo que "(No se refiere a una caja de cartón :)" sobraría.

vpedrosa commented on docs/gestores.md:
```diff
@@ -0,0 +1,73 @@
+
+Nuestras prioridades de cara a la gestión de dependencias son las siguientes:
+- Resulte fácil de instalar y usar para el programador; es decir, que en pocos pasos pueda disponer de las dependencias.
+- Aislamiento de los paquetes para evitar futuros errores 
```
Estoy de acuerdo con lo que te comenta @jorgelopez-ugr, además creo que podrías establecer otros criterios que además de más objetivos fueran más relevantes para el objetivo.

vpedrosa commented on docs/gestores.md:
```diff
@@ -0,0 +1,73 @@
+#### CPAN
+- Facilidad de instalación y uso puesto que viene con Perl. No requiere de pasos extra para su instalación.
+- Instala las dependencias de manera global, sin aislarlas respecto al proyecto.
+
```
Añade enlaces de la documentación oficial a las herramientas mencionadas, para que no tengamos que buscarlas.

vpedrosa commented on docs/gestores.md:
```diff
@@ -0,0 +1,73 @@
+Como CPAN, sirve solo para instalar dependencias
+- Fácil de usar, puesto que requiere de menos opciones el comando, que CPAN.
+- Requiere de instalación manual, pero se puede hacer fácilmente a través de CPAN 
+
```
Ídem enlaces.

vpedrosa commented on docs/gestores.md:
```diff
@@ -0,0 +1,73 @@
+- Aísla los paquetes para el proyecto
+- Se organiza todas las versiones y paquetes desde un mismo fichero:
+- Instalación fácil a través de CPAN y por debajo está usando cpan.
+
```
Ídem enlaces.

vpedrosa commented on docs/gestores.md:
```diff
@@ -0,0 +1,73 @@
+- Sencillo y fácil de instalar
+- Disponemos de Makefile.pl para perl y nos facilitaría el desarrollo
+- Herramienta muy estable
+
```
Ídem enlaces.

vpedrosa commented on docs/gestores.md:
```diff
@@ -0,0 +1,73 @@
+- Fácil de usar y de instalar
+- Nos permite gestionar las tareas para el desarrollo de cualquier tipo de proyecto.
+- La herramienta es estable y actualmente se encuentra en desarrollo activo.
+
```
Ídem enlaces.

vpedrosa commented on docs/gestores.md:
```diff
@@ -0,0 +1,73 @@
+#### Task
+- Tendríamos que instalar JS, para usarla, lo cual nos dificulta inicialmente en la instalación.
+- Y es un proyecto activo y estable.
+
```
Ídem enlaces.

vpedrosa commented on Makefile:
```diff
@@ -0,0 +1,5 @@
+	carton install
+
+check:
+	carton check
```
Esto no comprueba la sintaxis de los archivos .pm. Solo verifica que las dependencias del cpanfile estén instaladas. Fíjate que en la documentación del objetivo dice: "¿La tarea check efectivamente comprueba la sintaxis de los fuentes existentes y sólo eso?"

vpedrosa reviewed (changes requested):
Buenas @chelunike, te dejo una revisión de lo que he ido viendo, creo que te falta un poco aún para tener el objetivo completo.
